### ISSUE TITLE
Training Stuck at the pbar stage

### ISSUE THREAD

**Issue by harrywang-1523** [muhanzhang/IGMC#6]
I encounter a small problem during running this command: 
`python Main.py --data-name ml_10m --save-appendix _mnhp100 --data-appendix _mnph100 --max-nodes-per-hop 100 --testing --epochs 40 --save-interval 5 --adj-dropout 0 --lr-decay-step-size 20 --ensemble --dynamic-dataset`

The training just stuck here for hours:
<img width="719" alt="Screenshot 2020-07-10 at 9 27 20 AM" src="https://user-images.githubusercontent.com/24424127/87106254-99858e00-c28f-11ea-9df0-2f941d584353.png">

I made certain that Python ==**3.8.1** Pytorch==**1.4.0**, Torch Geometric == **1.4.2**. Not certain what is wrong. Thanks for your help in advance!

**Comment by muhanzhang** [comment]
Hi, IGMC needs to generate an independent enclosing subgraph for each rating. Because ML_10M is too large, the program is just waiting for the finish of the first epoch. You may check the cpu usage that it is indeed running.

To alleviate this issue, you may append "--max-train-num 100000" for example, so that IGMC only uses 100K ratings as training data. You may also append "--sample-ratio 0.2 --max-nodes-per-hop 50" for example to downsample each hop's neighbors to reduce subgraph sizes. These are the current remediations. But improving IGMC's scalability is the ultimate solution :)

**Comment by harrywang-1523** [comment]
Thank you very much for the prompt reply. If convenient, could you please give a rough estimation of how much computation is needed? For example, for ML_10M, what was your experiment environment and how long does it take?

**Comment by muhanzhang** [comment]
The paper did not use this dataset. I haven't fully run on this dataset too. If no downsampling is used, my estimate is it will take a week or two to finish. But the run time will vary significantly depending on your setting. You can always reduce "--sample-ratio", "--max-nodes-per-hop", "--max-train-num" or "--epochs" to reach an acceptable time. IGMC is relatively robust to these hyperparameters.

**Comment by harrywang-1523** [comment]
Okay. Thanks!
